compiler/rustc_mir_transform/src/simplify.rs

```diff
@@ -110,9 +111,13 @@ impl<'a, 'tcx> CfgSimplifier<'a, 'tcx> {
             }
         }
 
+        // Preserve `SwitchInt` reads on the phase transition from Built -> Analysis(Initial)
+        // or if `-Zmir-preserve-ub`.
+        let preserve_switch_reads =
+            matches!(body.phase, MirPhase::Built) | tcx.sess.opts.unstable_opts.mir_preserve_ub;
```

Comment:
I think we should preserve `SwitchInt` for *all* "analysis" invocations of this pass. Otherwise, further analyses like const_check or... not sure what else we have... might start depending on this optimization, which we definitely do not want.

Reply:
Also why are you using `|`, not `||`?